operator/controllers/clusterconfig_controller.go

```diff
@@ -47,15 +58,72 @@ type ClusterConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	reqLogger := log.WithValues("Request.Name", req.Name)
+	reqLogger.Info("reconciling ClusterConfig: START")
+	defer reqLogger.Info("reconciling ClusterConfig: STOP")
+
+	instance := &metalk8sscalitycomv1alpha1.ClusterConfig{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if req.Name == instanceName {
+				panic(fmt.Errorf(
+					"%s ClusterConfig object should not be deleted", req.Name,
+				))
+			}
+			reqLogger.Info("ClusterConfig already deleted: nothing to do")
+			return endReconciliation(nil)
+		}
+		reqLogger.Error(err, "cannot read ClusterConfig: requeue")
+		return delayedRequeue(err)
+	}
+
+	if instance.Name != instanceName {
+		if err := r.Delete(ctx, instance); err != nil {
+			reqLogger.Error(
+				err, "cannot delete extra ClusterConfig: requeue",
+			)
+			return delayedRequeue(err)
+		}
+		reqLogger.Info("deleting extra ClusterConfig, consider updating the default one")
+		return endReconciliation(nil)
+	}
 
 	// TODO(user): your logic here
 
-	return ctrl.Result{}, nil
+	return endReconciliation(nil)
+}
+
+// Trigger a reschedule after a short delay.
+func delayedRequeue(err error) (ctrl.Result, error) {
+	delay := 10 * time.Second
+	return ctrl.Result{Requeue: err == nil, RequeueAfter: delay}, err
```

Comment:
I did not tested this, this function comes from the storage-operator, from the documentation if `Requeue` is False he will not be requeued so I think no but I will do a test

Reply:
So I did some tests and research on the internet, and it appears that if there is an error in "Reconcile" return, it will be automatically requeued no matter the "Result" content AND the Controller automatically handle the retry backoff.

So I will update a bit the function